src/fromager/settings.py

```diff
@@ -143,16 +143,46 @@ def _resolve_template(
         raise
 
 
-def _parse(content: str) -> Settings:
-    data = yaml.safe_load(content)
-    return Settings(data)
+def load(settings_file: pathlib.Path, settings_dir: pathlib.Path) -> Settings:
+    settings_data = {}
 
-
-def load(filename: pathlib.Path) -> Settings:
-    filepath = pathlib.Path(filename)
+    filepath = pathlib.Path(settings_file)
     if not filepath.exists():
         logger.debug("settings file %s does not exist, ignoring", filepath.absolute())
-        return Settings({})
-    with open(filepath, "r") as f:
-        logger.info("loading settings from %s", filepath.absolute())
-        return _parse(f.read())
+    else:
+        with open(filepath, "r") as f:
+            logger.info("loading settings from %s", filepath.absolute())
+            settings_data = yaml.safe_load(f.read())
+
+    # Per-package files are inserted in the `packages` key using the name that
+    # will be used to look the value up. Transform any existing keys to that
+    # format so we can warn if there are overriding values.
+    package_settings_from = {}
+    if "packages" not in settings_data:
+        settings_data["packages"] = {}
+    else:
+        new_packages = {}
+        for name, value in settings_data["packages"].items():
+            package_name = overrides.pkgname_to_override_module(name)
+            new_packages[package_name] = value
+            package_settings_from[package_name] = settings_file
+
+    for package_file in sorted(settings_dir.glob("*.yaml")):
```

Comment:
How do you feel about a separate directory for package settings like `settings.d`, `packages.d` or just `packages`?
```suggestion
    for package_file in sorted(settings_dir.glob("packages.d/*.yaml")):
```

Reply:
I called it just `settings` to be consistent with our other names, like `env`.

I could see us eventually removing this global file and moving the pre-built flag into these files.